packages/client/hmi-client/src/components/operator/tera-operator-output-summary.vue

```diff
@@ -3,98 +3,129 @@
 			</div>
 		</template>
-		<div v-else-if="!isNil(activeOutput?.summary)" class="summary">
-			<img v-if="isGenerating || isGenerated" src="@assets/svg/icons/magic.svg" alt="Magic icon" />
-			<p v-if="isGenerating">Generating AI summary...</p>
-			<p v-else @click="isEditing = true">
-				{{ summary }}<span class="pi pi-pencil ml-2 text-xs" />
+		<div v-else-if="isLoading === false" class="summary">
+			<p else @click="isEditing = true">
+				{{ summaryText }}<span class="pi pi-pencil ml-2 text-xs" />
 			</p>
 		</div>
+		<div v-else-if="isLoading" class="summary">
+			<img src="@assets/svg/icons/magic.svg" alt="Magic icon" />
+			<p>Generating summary...</p>
+		</div>
 	</section>
 </template>
 
 <script setup lang="ts">
-import { cloneDeep, isNil } from 'lodash';
-import { ref, watch, PropType, computed } from 'vue';
+import { isEmpty } from 'lodash';
+import { ref, watch, onUnmounted } from 'vue';
+import { Poller, PollerState } from '@/api/api';
 import Textarea from 'primevue/textarea';
 import Button from 'primevue/button';
-import { getActiveOutput } from '@/services/workflow';
-import { WorkflowNode } from '@/types/workflow';
+import { updateSummary, getSummaries } from '@/services/summary-service';
+import { Summary } from '@/types/Types';
 
 const MAX_LENGTH = 400;
 
 const props = defineProps({
-	node: {
-		type: Object as PropType<WorkflowNode<any>>,
+	summaryId: {
+		type: String,
 		required: true
 	}
 });
 
-const activeOutput = computed(() => getActiveOutput(props.node));
-
-const emit = defineEmits(['update-output-port', 'generate-output-summary']);
-
-const summary = ref(activeOutput.value?.summary);
-
+const summaryText = ref('');
+const summary = ref<Summary | null>(null);
 const isEditing = ref(false);
+const isLoading = ref(true);
 
-function updateSummary() {
-	const updated = cloneDeep(activeOutput.value);
-	if (!updated) return;
-	if (updated.summary === summary.value) {
-		isEditing.value = false;
-		return;
-	}
-	updated.summaryHasBeenEdited = true;
-	updated.summary = summary.value;
-	emit('update-output-port', updated);
+function updateSummaryText() {
+	if (!summary.value || isEmpty(summary.value)) return;
+	summary.value.humanSummary = summaryText.value;
+	updateSummary(summary.value);
 	isEditing.value = false;
 }
 
+function getSummaryText(s: Summary) {
+	if (s.humanSummary) {
+		return s.humanSummary;
+	}
+	return s.generatedSummary || '';
+}
+
 function cancelEdit() {
 	isEditing.value = false;
-	summary.value = activeOutput.value?.summary;
+	if (summary.value) {
+		summaryText.value = getSummaryText(summary.value);
+	}
 }
 
-const isGenerating = computed(
-	() => activeOutput?.value?.summaryHasBeenEdited !== true && activeOutput?.value?.summary === ''
-);
-const isGenerated = computed(
-	() =>
-		activeOutput?.value?.summaryHasBeenEdited !== true &&
-		(activeOutput?.value?.summary?.length ?? 0) > 0
-);
+/**
+ * It can take some time (30-40 seconds) for generated summary to appear
+ * */
```

Comment:
Because the summary is done via the gollm-taskrunner, why not using the notifications?

Reply:
Notification should be the end-product of some long running process, it should not be used as a bridge. For example I think the following model would be incorrect:

```
Start generating summary ==> [Notification bus] ==> Update summary
```

This can introduce many problems like a self perpetuating loop, and a bit harder to tracking thigns.

Reply:
Instead of polling, you can listen for the SEE event for the task response update which isn't necessarily same as the notification. 
So it would be like
```
After generating summary, 
1. Try fetching summary ==> if not ready, listen for SSE event ==> on success event, fetch summary object again ( or use output from the task response)  
2.  Try fetching summary ==> if ready, just use the summary. 
```
But I think in this case, ^ requires more logic with a bit more complexity than just doing simple polling.

Reply:
@jryu01 you may have just volunteered yourself

There are a number of edge cases on top of what you had mentioned, technically speaking you can invoke the next run before the first run had finished generating a summary, so you end up tracking multiple signals. 
- Invoke run ... call it A
- A's summary is generating
- Invoke 2nd run, call it B
- B's summary is generating
- B summary finishes
- A summary finishes